example_packages/profiles_priorities/d1/fpm.toml

```diff
@@ -0,0 +1,21 @@
+name = "d1"
+
+[library]
+source-dir="source"
+
+[[executable]]
+name="d1"
+source-dir="app"
+main="d1.f90"
+
+[profiles.debug.gfortran]
+flags="-g -O1"
+
+[profiles.debug.gfortran.windows]
+flags="/g /O1"
+
+[dependencies]
+"d11" = {path = "../d11"}
+"d12" = {path = "../d12"}
```

Comment:
Instead  of using compiler flags here, how about setting preprocessor defines for the same variable in each package with different values. Each package can than check whether it has the expected value.

Reply:
I have tried to implement this, but I am struggling on forcing fpm to preprocess the code. When I changed the file extensions to .F90 (all the source files and app/main.F90 in main_package), it says:
<ERROR>*cmd_build*:package error:Neither library nor executable found, there is nothing to do
Am I missing something?

Reply:
Is the new test more stable @awvwgk ?